src/test/java/core/basesyntax/DaoTest.java

```diff
@@ -0,0 +1,126 @@
+        reportGenerator = new ReportGeneratorImpl(fruitStorage);
+    }
+
+    @Test
+    void read_validFile_Ok() {
+        List<String> result = fileReader.read("src/test/java/core/basesyntax/testfile.csv");
+        String joinedResult = String.join(", ", result);
+        assertEquals("BALANCE,apple,100, SUPPLY,banana,50", joinedResult);
+    }
+
+    @Test
+    void read_nonExistentFile_NotOk() {
+        assertThrows(RuntimeException.class, () -> fileReader
+                .read("nonexistentfolder/nonexistentfile.csv"));
+    }
+
+    @Test
+    void writeToFile_ok() {
+        String expectedContent = "BALANCE,apple,100, SUPPLY,banana,50";
+        String filePath = "src/test/java/core/basesyntax/validfile.csv";
+        fileWriter.write(expectedContent, filePath);
+
+        List<String> result = fileReader.read(filePath);
+        String joinedResult = String.join(", ", result);
+        assertEquals("BALANCE,apple,100, SUPPLY,banana,50", joinedResult);
+    }
+
+    @Test
+    void write_toNonExistentFile_NotOk() {
+        assertThrows(RuntimeException.class, () -> fileWriter
+                .write("BALANCE,apple,100, SUPPLY,banana,50",
+                        "nonexistentfolder/nonexistentfile.csv"));
+    }
+
+    @Test
+    void convertData_Ok() {
+        List<String> report = Arrays.asList("BALANCE,apple,100", "SUPPLY,banana,50");
+        List<FruitTransaction> result = dataConverter.convertToTransaction(report);
+
+        assertEquals(2, result.size());
+        assertEquals(new FruitTransaction(Operation.BALANCE, "apple", 100), result.get(0));
+        assertEquals(new FruitTransaction(Operation.SUPPLY, "banana", 50), result.get(1));
+    }
+
+    @Test
+    void convert_toWrongFormat_notOk() {
+        List<String> report = Arrays.asList("BALANCE,apple", "SUPPLY,banana,50");
+
+        assertThrows(IllegalArgumentException.class,
+                () -> dataConverter.convertToTransaction(report));
+    }
+
+    @Test
+    void convert_illegalOperation_notOK() {
+        List<String> report = Arrays.asList("NONEXISTENTOPERATION,apple,100", "SUPPLY,banana,50");
+
+        assertThrows(IllegalArgumentException.class,
+                () -> dataConverter.convertToTransaction(report));
+    }
+
+    @Test
+    void convert_illegalQuantity_notOK() {
+        List<String> report = Arrays.asList("BALANCE,apple,-20", "SUPPLY,banana,50");
+
+        assertThrows(IllegalArgumentException.class,
+                () -> dataConverter.convertToTransaction(report));
+    }
+
+    @Test
+    void convert_illegalQuantityFormat_notOK() {
+        List<String> report = Arrays.asList("BALANCE,apple,20.00", "SUPPLY,banana,50");
+
+        assertThrows(IllegalArgumentException.class,
+                () -> dataConverter.convertToTransaction(report));
+    }
+
+    @Test
+    void generateReport_Ok() {
+        fruitStorage.addFruit("apple", 100);
+        fruitStorage.addFruit("banana", 50);
+        fruitStorage.addFruit("peach", 10);
```

Comment:
Since the Fruit Map in your FruitStorage simulates a real database, it's best to make it static. This will provide a better understanding for writing integration tests in the future.
Don’t forget to clear it after each test ;)